flocker/acceptance/test_api.py

```diff
@@ -369,6 +369,57 @@ def can_connect():
         d.addCallback(check_result)
         return d
 
+    def test_create_container_with_environment(self):
+        """
+        Create a container including environment variables on a single-node
+        cluster.
+        """
+        data = {
+            u"name": "my_env_container",
+            u"host": None,
+            u"image": "busybox:latest",
+            u"environment": {u"ACCEPTANCE_ENV_LABEL": 'acceptance test ok'}
+        }
+        waiting_for_cluster = wait_for_cluster(test_case=self, node_count=1)
+
+        def create_container(cluster, data):
+            data[u"host"] = cluster.nodes[0].address
+            return cluster.create_container(data)
+
+        d = waiting_for_cluster.addCallback(create_container, data)
+
+        def check_result(result):
+            response = result[1]
+
+            def can_connect():
+                try:
+                    remote_data = run_SSH(
+                        22, 'root', data[u"host"],
+                        [
+                            b"docker",
+                            b"inspect",
+                            b"flocker--{0}".format(data[u"name"])
+                        ], None
+                    )
+                except Exception:
+                    return False
+                container_data = loads(remote_data)
+                env_data = container_data[0]['Config']['Env']
+                for var in env_data:
+                    try:
+                        label, value = var.split("=")
+                        if [(label, value)] == data[u"environment"].items():
+                            return True
+                    except ValueError:
+                        continue
+                return False
+            dl = loop_until(can_connect)
+            self.assertEqual(response, data)
```

Comment:
No, this assertion doesn't rely on anything that happens in the `loop_until` call; this is just checking the API response matches up to the supplied and expected JSON correctly. The separate looping call ensures that the test will either succeed when it can connect via SSH and inspect the container and finds a match to the injected environment variable keys and values, or the whole test will just continuously run until it times out through trial.